src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,50 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl<T extends Machine> implements MachineService<T> {
+    @Override
+    public List<T> getAll(Class<? extends T> type) {
+        MachineProducer<T> producer = null;
+        if (type == Bulldozer.class) {
+            producer = (MachineProducer<T>) new BulldozerProducer();
+        }
+        if (type == Truck.class) {
+            producer = (MachineProducer<T>) new TruckProducer();
+        }
+        if (type == Excavator.class) {
+            producer = (MachineProducer<T>) new ExcavatorProducer();
+        }
+        if (producer != null) {
+            return (List<T>) producer.get();
+        } else {
```

Comment:
Keep if-else constructions simple. Since you are returning inside the if block, there is no need for an else block. You can directly return after the if block.